workers/data_refinery_workers/processors/salmon.py

```diff
@@ -0,0 +1,258 @@
+        job_context["kmer_size"] = "31"
+    else:
+        job_context["kmer_size"] = "23"
+
+    return job_context
+
+
+def _download_index(job_context: Dict) -> Dict:
+    batch = job_context["batches"][0]
+    try:
+        index_batches = Batch.objects.filter(source_type=Downloaders.TRANSCRIPTOME_INDEX.value,
+                                             organism_id=batch.organism_id,
+                                             status=BatchStatuses.PROCESSED.value).all()
+        batch_ids = [batch.id for batch in index_batches]
+
+        batch_key_value = BatchKeyValue.objects.filter(batch_id__in=batch_ids,
+                                                       key="kmer_size",
+                                                       value=job_context["kmer_size"]).all()[0]
+
+        index_batch = batch_key_value.batch
+        index_file = index_batch.files[0]
+    except:
+        logger.exception("Failed to find an index for organism %s with kmer_size of %s.",
+                         batch.organism_name,
+                         job_context["kmer_size"],
+                         processor_job=job_context["job_id"],
+                         batch=batch.id)
+
+        failure_template = "Failed to find an index for organism {} with kmer_size of {}."
+        job_context["job"].failure_reason = failure_template.format(batch.organism_name,
+                                                                    job_context["kmer_size"])
+        job_context["success"] = False
+        return job_context
+
+    temp_dir = batch.files[0].get_temp_dir(job_context["job_dir_prefix"])
+    job_context["index_directory"] = os.path.join(temp_dir, "index")
+    os.makedirs(job_context["index_directory"], exist_ok=True)
+    try:
+        # The index_batch file will be downloaded to a directory
+        # specific to that batch.
+        temp_path = index_file.download_processed_file(job_context["job_dir_prefix"])
+        with tarfile.open(temp_path, "r:gz") as tarball:
+            tarball.extractall(temp_dir)
+    except:
+        logger.exception("Failed to download and extract index tarball %s",
+                         index_file.name,
+                         index_batch=index_batch.id,
+                         index_file=index_file.id,
+                         processor_job=job_context["job_id"],
+                         batch=batch.id)
+
+        failure_template = "Failed to download and extract index tarball {}"
+        job_context["job"].failure_reason = failure_template.format(index_file.name)
+        job_context["success"] = False
+        return job_context
+    finally:
+        # Remove temp dir for the index batch since we've extracted it to
+        # the temp dir for this job.
+        index_file.remove_temp_directory(job_context["job_dir_prefix"])
+
+    return job_context
+
+
+def _zip_and_upload(job_context: Dict) -> Dict:
+    # If there are paired reads... the file name will be based off of
```

Comment:
It would be good to have these comments in the Python docstring format `""" """`, especially in case we ever want to export the documentation for all of the functions (or access it from `__doc__` or `help()`).

Reply:
I agree I should add a docstring here, but these comments are not about the function, they're about the 4 lines of code they immediately precede.